lib/datadog/tracing/metadata/meta_struct.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Tracing
+    module Metadata
+      # Adds complex structures tagging behavior through meta_struct
+      # @public_api
+      module MetaStruct
+        def set_meta_struct(meta_struct)
+          self.meta_struct.merge!(meta_struct)
+        end
+
+        def get_meta_struct(key)
```

Comment:
Now sure about naming, but if I call `get_meta_struct` I would expect the whole meta-struct object in return. I think we should somehow highlight that it's a part inside the meta-struct.

Maybe names like

```
get_meta_struct_value
get_meta_struct_field
```

or something way better (see ⬆️)

Reply:
why not just `def [](key)`?